### PR TITLE
Quick Sort Implementation in Kotlin

### LINKED ISSUE #2131: Quick Sort
- [x] C
- [x] C++
- [x] C# - @Yashank18
- [ ] CoffeeScript - @prashansag62
- [x] Dart
- [x] Go
- [x] Java
- [x] JS
- [x] Kotlin - @97shivank
- [x] Php - @goyalmuskan
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust - @m-code12
- [ ] TypeScript - @dj373

### PR DESCRIPTION
**Pull request by 97shivank**
Fixes #2131 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- implementation of quicksort algorithm in kotlin.

#### Languages Used:
- Kotlin

#### Files Added:
- Quick_Sort.kt

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by 97shivank** [comment]
@jainaman224 kindly review this pr

File: Quick_Sort/Quick_Sort.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    exchange(A, i + 1, r)
+    return i + 1
+}
+//swapping
+fun exchange(A: Array<Int>, i: Int, j: Int) {
+    var temp = A[i]
+    A[i] = A[j]
+    A[j] = temp
+}
+//main
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:29]: This comment doesn't add any value to main function. You can remove it.

File: Quick_Sort/Quick_Sort.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            i++
+            exchange(A, i, j)
+        }
+    }
+    exchange(A, i + 1, r)
+    return i + 1
+}
+//swapping
+fun exchange(A: Array<Int>, i: Int, j: Int) {
+    var temp = A[i]
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:25]: It would be best to call this function swap if it is not a keyword in kotlin

File: Quick_Sort/Quick_Sort.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    for (j in p until r) {
+        if (A[j] <= x) {
+            i++
+            exchange(A, i, j)
+        }
+    }
+    exchange(A, i + 1, r)
+    return i + 1
+}
+//swapping
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:23]: Please leave a blank line before this.

File: Quick_Sort/Quick_Sort.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
+//partition array
+fun quick_sort(A: Array<Int>, p: Int, r: Int) {
+    if (p < r) {
+        var q: Int = partition(A, p, r)
+        quick_sort(A, p, q - 1)
+        quick_sort(A, q + 1, r)
+
+    }
+}
+//assign last value as pivot
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:10]: Please leave a blank line before this.

File: Quick_Sort/Quick_Sort.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
+//partition array
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:1]: Please add a brief description of quick sort. 1-2 lines.

File: Quick_Sort/Quick_Sort.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    return i + 1
+}
+
+//swap
+fun exchange(A: Array<Int>, i: Int, j: Int) {
+    var temp = A[i]
+    A[i] = A[j]
+    A[j] = temp
+}
+fun main(arg: Array<String>) {
```
Comment (singh-shreya6) [inline Quick_Sort/Quick_Sort.kt:34]: Please leave one blank line before this.

**Comment by 97shivank** [comment]
@singh-shreya6 done kindly look into it .

**Pull request merged**